src/raw_upload.rs

```diff
@@ -0,0 +1,101 @@
+use base64::prelude::*;
+use pyo3::prelude::*;
+use std::collections::HashSet;
+use std::io::prelude::*;
+
+use flate2::bufread::ZlibDecoder;
+
+use quick_xml::reader::Reader;
+use serde::Deserialize;
+
+use crate::junit::{get_position_info, use_reader};
+use crate::testrun::ParsingInfo;
+use crate::ParserError;
+
+#[derive(Deserialize, Debug, Clone)]
+struct TestResultFile {
+    filename: String,
+    #[serde(skip_deserializing)]
+    _format: String,
+    data: String,
+    #[serde(skip_deserializing)]
+    _labels: Vec<String>,
+}
+#[derive(Deserialize, Debug, Clone)]
+struct RawTestResultUpload {
+    #[serde(default)]
+    network: Option<Vec<String>>,
+    test_results_files: Vec<TestResultFile>,
+}
+
+#[derive(Debug, Clone)]
+struct ReadableFile {
+    filename: Vec<u8>,
+    data: Vec<u8>,
+}
+
+const LEGACY_FORMAT_PREFIX: &[u8] = b"# path=";
+const LEGACY_FORMAT_SUFFIX: &[u8] = b"<<<<<< EOF";
+
+fn serialize_to_legacy_format(readable_files: Vec<ReadableFile>) -> Vec<u8> {
+    let mut res = Vec::new();
+    for file in readable_files {
+        res.extend_from_slice(LEGACY_FORMAT_PREFIX);
+        res.extend_from_slice(&file.filename);
+        res.extend_from_slice(b"\n");
+        res.extend_from_slice(&file.data);
+        res.extend_from_slice(b"\n");
+        res.extend_from_slice(LEGACY_FORMAT_SUFFIX);
+        res.extend_from_slice(b"\n");
+    }
+    res
+}
+
+#[pyfunction]
+#[pyo3(signature = (raw_upload_bytes))]
+pub fn parse_raw_upload(raw_upload_bytes: &[u8]) -> PyResult<(Vec<u8>, Vec<u8>)> {
+    let upload: RawTestResultUpload = serde_json::from_slice(raw_upload_bytes)
+        .map_err(|e| ParserError::new_err(format!("Error deserializing json: {}", e)))?;
```

Comment:
do you think our custom `ParserError` is pulling its weight? pyo3 has support for `anyhow`, which might be simpler to use than this custom error type.

With anyhow, this would just be `.context("deserializing raw upload JSON")?`